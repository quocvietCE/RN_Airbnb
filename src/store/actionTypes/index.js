export const AUTH_ACTION = {
  USER_LOGIN: 'USER_LOGIN',
  USER_LOGIN_SUCCESS: 'USER_LOGIN_SUCCESS',
  USER_LOGIN_FAILURE: 'USER_LOGIN_FAILURE',

  LOG_OUT: 'LOG_OUT',
};

export const APP_ACTION = {
  UPDATE_LANGUAGE: 'UPDATE_LANGUAGE',
  REQUIRE_LOGIN_PAGE: 'REQUIRE_LOGIN_PAGE',
  UPDATE_POSITION: 'UPDATE_POSITION',
  OPEN_MODAL: 'OPEN_MODAL',
  CLOSE_MODAL: 'CLOSE_MODAL',
  OPEN_ADDRESS_MODAL: 'OPEN_ADDRESS_MODAL',
  CLOSE_ADDRESS_MODAL: 'CLOSE_ADDRESS_MODAL',
  CHANGED_COUNTRY: 'CHANGED_COUNTRY',
  CHANGE_COUNTRY_SUCCESS: 'CHANGE_COUNTRY_SUCCESS',
  SET_APP_DATA: 'SET_APP_DATA',
  GET_CONFIGURATION_APP: 'GET_CONFIGURATION_APP',
  GET_CONFIGURATION_APP_SUCCESS: 'GET_CONFIGURATION_APP_SUCCESS',
  GET_CONFIGURATION_APP_FAIL: 'GET_CONFIGURATION_APP_FAIL',
  CLEAR_ERROR: 'CLEAR_ERROR',
  OPEN_FULL_MAP_VIEW: 'OPEN_FULL_MAP_VIEW',
  CLOSE_FULL_MAP_VIEW: 'CLOSE_FULL_MAP_VIEW',
  SAVE_PERMISSION_PHOTO: 'SAVE_PERMISSION_PHOTO',
};

export const LISTING_ACTION = {
  GET_LISTING_DATA: 'GET_LISTING_DATA',
  GET_LISTING_DATA_SUCCESS: 'GET_LISTING_DATA_SUCCESS',
  GET_LISTING_DATA_FAILURE: 'GET_LISTING_DATA_FAILURE',
};

export const NETWORK_ACTION = {
  SET_STATUS_CONNECTED: 'SET_STATUS_CONNECTED',
  SET_STATUS_CONNECTED_SUCCESS: 'SET_STATUS_CONNECTED_SUCCESS',
  SET_STATUS_CONNECTED_FAILURE: 'SET_STATUS_CONNECTED_FAILURE',
};
