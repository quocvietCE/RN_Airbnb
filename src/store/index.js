import {createStore, applyMiddleware, compose} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {persistStore, persistReducer} from 'redux-persist';
import AsyncStorage from '@react-native-community/async-storage';
import logger from 'redux-logger';
import rootSaga from './sagas';

import rootReducer from './reducers';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  // blacklist: ['auth', 'app'],
  blacklist: ['app, listing'],
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const sagaMiddleware = createSagaMiddleware();

export const store = createStore(
  persistedReducer,
  composeEnhancer(applyMiddleware(sagaMiddleware, logger)),
);

export const persistor = persistStore(store);

sagaMiddleware.run(rootSaga);
export default () => ({store, persistor});
