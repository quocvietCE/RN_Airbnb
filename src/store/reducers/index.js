import {combineReducers} from 'redux';
import auth from './authReducer';
import app from './appReducer';
import listing from './listingReducer';
import network from './networkReducer';

export default combineReducers({
  auth,
  app,
  listing,
  network,
});
