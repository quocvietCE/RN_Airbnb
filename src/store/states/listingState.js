const initState = {
  listingData: [],
  favouriteListings: [],
};

const listingState = {
  initState,
};

export default listingState;
