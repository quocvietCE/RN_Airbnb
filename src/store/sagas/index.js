import {takeLatest, all, takeEvery} from 'redux-saga/effects';
import authSaga from './authSaga';
import {AUTH_ACTION} from '../actionTypes';

function* handleLogin() {
  yield takeLatest(AUTH_ACTION.USER_LOGIN, authSaga.handleLogin);
}

export default function* rootSaga() {
  yield all([handleLogin()]);
}
