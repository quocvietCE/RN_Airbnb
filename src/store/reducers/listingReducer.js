import {LISTING_ACTION} from '../actionTypes';
import listingState from '../states/listingState';

const listingReducer = (state = listingState.initState, action) => {
  switch (action.type) {
    case LISTING_ACTION.GET_LISTING_DATA_SUCCESS:
      return {
        ...state,
        listingData: action.result || [],
      };
    default:
      return state;
  }
};

export default listingReducer;
