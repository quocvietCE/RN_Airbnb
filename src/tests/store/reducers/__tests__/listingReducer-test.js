import React from 'react';
import 'react-native';
import listingReducer from '../../../../store/reducers/listingReducer';
import listingState from '../../../../store/states/listingState';
import {LISTING_ACTION} from '../../../../store/actionTypes';

describe('Test listingReducer', () => {
  it('returns the same listingReducer state on an unhandled state', () => {
    expect(
      listingReducer(listingState.initState, {type: '_NULL'}),
    ).toMatchSnapshot();
  });

  it('handles get request LISTING_ACTION.GET_LISTING_DATA_SUCCESS', () => {
    expect(
      listingReducer(listingState.initState, {
        type: LISTING_ACTION.GET_LISTING_DATA_SUCCESS,
      }),
    ).toEqual({
      ...listingState.initState,
      listingData: [],
    });
  });
});
