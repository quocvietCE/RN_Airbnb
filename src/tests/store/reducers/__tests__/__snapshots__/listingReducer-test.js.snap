// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test listingReducer returns the same listingReducer state on an unhandled state 1`] = `
Object {
  "favouriteListings": Array [],
  "listingData": Array [],
}
`;
